specs/_features/eip7547/beacon-chain.md

```diff
@@ -0,0 +1,273 @@
+    - [Engine APIs](#engine-apis)
+      - [New `verify_and_notify_new_inclusion_list`](#new-verify_and_notify_new_inclusion_list)
+      - [Modified `verify_and_notify_new_payload`](#modified-verify_and_notify_new_payload)
+  - [Block processing](#block-processing)
+    - [Execution payload](#execution-payload)
+      - [Modified `process_execution_payload`](#modified-process_execution_payload)
+      - [New `verify_inclusion_list_summary_signature`](#new-verify_inclusion_list_summary_signature)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This is the beacon chain specification to add an inclusion list mechanism to allow forced transaction inclusion. Refers to [EIP-7547](https://eips.ethereum.org/EIPS/eip-7547).
+
+*Note:* This specification is built upon [Deneb](../../deneb/beacon_chain.md) and is under active development.
+
+## Constants
+
+### Domain types
+
+| Name | Value |
+| - | - |
+| `DOMAIN_INCLUSION_LIST_SUMMARY`     | `DomainType('0x0B000000')` |
+
+## Preset
+
+### Execution
+
+| Name | Value |
+| - | - |
+| `MAX_TRANSACTIONS_PER_INCLUSION_LIST` |  `uint64(16384) = 2**14` |
+
+## Containers
+
+### New Containers
+
+#### `InclusionListSummaryEntry`
+
+```python
+class InclusionListSummaryEntry(Container):
+    address: ExecutionAddress
+    nonce: uint64
+```
+
+#### `InclusionListSummary`
+
+```python
+class InclusionListSummary(Container):
+    slot: Slot
+    proposer_index: ValidatorIndex
+    parent_hash: Hash32
+    summary: List[InclusionListSummaryEntry, MAX_TRANSACTIONS_PER_INCLUSION_LIST]
+```
+
+#### `SignedInclusionListSummary`
+
+```python
+class SignedInclusionListSummary(Container):
+    message: InclusionListSummary
+    signature: BLSSignature
+```
+
+#### `InclusionList`
+
+```python
+class InclusionList(Container):
+    signedSummary: SignedInclusionListSummary
+    transactions: List[Transaction, MAX_TRANSACTIONS_PER_INCLUSION_LIST]
+```
+
+### Extended containers
+
+#### `ExecutionPayload`
+
+Adding `previous_inclusion_list_summary`.
+
+```python
+class ExecutionPayload(Container):
+    # Execution block header fields
+    parent_hash: Hash32
+    fee_recipient: ExecutionAddress  # 'beneficiary' in the yellow paper
+    state_root: Bytes32
+    receipts_root: Bytes32
+    logs_bloom: ByteVector[BYTES_PER_LOGS_BLOOM]
+    prev_randao: Bytes32  # 'difficulty' in the yellow paper
+    block_number: uint64  # 'number' in the yellow paper
+    gas_limit: uint64
+    gas_used: uint64
+    timestamp: uint64
+    extra_data: ByteList[MAX_EXTRA_DATA_BYTES]
+    base_fee_per_gas: uint256
+    # Extra payload fields
+    block_hash: Hash32  # Hash of execution block
+    transactions: List[Transaction, MAX_TRANSACTIONS_PER_PAYLOAD]
+    withdrawals: List[Withdrawal, MAX_WITHDRAWALS_PER_PAYLOAD]
+    blob_gas_used: uint64
+    excess_blob_gas: uint64
+    previous_inclusion_list_summary: InclusionListSummary  # [New in EIP7547]
```

Comment:
Nit: Maybe we should rename it from `previous_inclusion_list_summary` to `parent_inclusion_list_summary` to align better

Reply:
done! parent here and in the ExecutionPayloadHeader too :D